devel/libbsd/Portfile

```
@@ -0,0 +1,117 @@
+
+    known_fail      yes
+
+    version         0.11.7
+
+    revision        0
+
+    master_sites    https://libbsd.freedesktop.org/releases/
+
+    use_xz          yes
+
+    checksums       rmd160  76c606f4e2c6bfce1fe9fff221de23a14b6887a7 \
+                    sha256  9baa186059ebbf25c06308e9f991fda31f7183c0f24931826d83aa6abd8a0261 \
+                    size    418508
+
+    conflicts       libbsd-devel
+
+} else {
+
+    # This is the libbsd-devel subport for Development
+
+    long_description {*}${long_description} \
+        This port provides the devel version of ${name}, as appropriate.
+
+    version         0.11.7-dirty
+
+    revision        0
+
+    # Development subport and we are using a git branch to build.  For example, with:
+    # distname        21d12b02112097f0c195dceb1892c95b7b957b36
+    # We would fetch the file archive from:
+    # "https://gitlab.freedesktop.org/libbsd/libbsd/-/archive/ ... \
+    #     21d12b02112097f0c195dceb1892c95b7b957b36/libbsd-21d12b02112097f0c195dceb1892c95b7b957b36.tar.bz2"
+
+    distname        73b25a8f871b3a20f6ff76679358540f95d7dbfd
+    master_sites    https://gitlab.freedesktop.org/libbsd/libbsd/-/archive/${distname}
+    distfiles       ${name}-${distname}.tar.bz2
+
+    use_bzip2       yes
+
+    checksums       rmd160  50e567fa91f0e019c55a9f0ef8b6d2ec904cebd7 \
+                    sha256  b498bd46a85e471e74ad3c20dc3045f02f8d345babf59ccec9d10f57a6a509d3 \
+                    size    135723
+
+    worksrcdir      ${name}-${distname}
+
+    conflicts       libbsd
+
+    patchfiles      patch010-configure.ac.diff \
+                    patch015-src_Makefile.am.diff \
+                    patch016-src_Makefile.am_libbsd.sym.diff \
+                    patch017-test_Makefile.am.diff \
+                    patch018-man_Makefile.am.diff
+
+    # For a git branch the file '.dist-version' will be missing, so we make one.
+    # Otherwise we would get an error from a script.  Currently, no need to check the file.
+    pre-configure   {
+                    set outfile [open "${worksrcpath}/.dist-version" w]
+                    puts $outfile "${version}_${revision}"
+                    close $outfile
+                    }
+
+    configure.cmd   ./autogen && ./configure
+
+    # We need to do a 'make check' for running the tests.
+    test.run        yes
+    test.target     check
+
+notes "
+This build of ${name} was derived from a GitLab branch and is therefore to be used with\
+care and might be unstable.\
+The relevant branch is: '${distname}'\
+\n
+CAVEAT: Currently, the utility 'closefrom' would fail the tests and should NOT be used\
+for building other programs.
+"
+
+}
+
+depends_build    port:libtool \
+                 port:pkgconfig \
+                 port:automake \
+                 port:autoconf
+
+depends_lib      port:libmd
+
+livecheck.type   regex
+livecheck.url    https://libbsd.freedesktop.org/releases/
+livecheck.regex  libbsd-(\[0-9.\]+)\.tar\.xz
```

You need double backslashes to escape the periods. The first backslash escapes from Tcl; the second escapes from the regular expression engine:

```suggestion
livecheck.regex  libbsd-(\[0-9.\]+)\\.tar\\.xz
```

Or better yet, since you're not interpolating any variables or procedures, use `{` curly brace quoting `}` to eliminate the need for one level of escaping:

```suggestion
livecheck.regex  {libbsd-([0-9.]+)\.tar\.xz}
```

---

Good point. Thank you very much for the good advice.